source/common/http/http2/codec_impl.cc

```diff
@@ -114,13 +114,14 @@ void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
 }
 
 ssize_t ConnectionImpl::StreamImpl::onDataSourceRead(size_t length, uint32_t* data_flags) {
-  if (pending_send_data_.length() == 0 && !local_end_stream_) {
+  size_t pending_send_data_length = pending_send_data_.length();
```

Comment:
uint64_t: length() returns that and we are trying to make sure it is 64-bit unsigned (prefer to not use size_t for the reasons you are trying to fix).